spark/src/test/java/org/apache/iceberg/spark/data/TestSparkAvroReaderForFieldsWithDefaultValue.java

```diff
@@ -50,72 +52,150 @@ public class TestSparkAvroReaderForFieldsWithDefaultValue {
+            "        \"name\": \"F5\",\n" +
+            "        \"fields\" : [\n" +
+            "          {\"name\": \"ff1\", \"type\": \"long\"},\n" +
+            "          {\"name\": \"ff2\", \"type\":  \"string\"}\n" +
+            "        ]\n" +
+            "      },\n" +
+            "      \"default\": {\n" +
+            "        \"ff1\": 999,\n" +
+            "        \"ff2\": \"foo\"\n" +
+            "      }\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"name\": \"f6\",\n" +
+            "      \"type\": {\n" +
+            "        \"type\": \"map\",\n" +
+            "        \"values\": {\n" +
+            "          \"type\": \"array\",\n" +
+            "          \"items\" : \"int\"\n" +
+            "        }\n" +
+            "      },\n" +
+            "      \"default\": {\"key\": [1, 2, 3]}\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"name\": \"f7\",\n" +
+            "      \"type\": {\n" +
+            "        \"type\": \"fixed\",\n" +
+            "        \"name\": \"md5\",\n" +
+            "        \"size\": 2\n" +
+            "      },\n" +
+            "      \"default\": \"FF\"\n" +
+            "    }\n" +
+            "  ]\n" +
+            "}";
+    org.apache.avro.Schema evolvedSchema = new org.apache.avro.Schema.Parser().parse(evolvedSchemaString);
 
-    // read written rows with evolved schema
     List<InternalRow> rows;
     Schema icebergReadSchema = AvroSchemaUtil.toIceberg(evolvedSchema);
+    // read written rows with evolved schema
     try (AvroIterable<InternalRow> reader = Avro.read(Files.localInput(testFile))
-        .createReaderFunc(SparkAvroReader::new)
-        .project(icebergReadSchema)
-        .build()) {
+            .createReaderFunc(SparkAvroReader::new)
+            .project(icebergReadSchema)
+            .build()) {
       rows = Lists.newArrayList(reader);
     }
 
-    // validate all rows, and all fields are read properly
     Assert.assertNotNull(rows);
     Assert.assertEquals(expected.size(), rows.size());
     for (int row = 0; row < expected.size(); row++) {
-      GenericData.Record expectedRow = expected.get(row);
       InternalRow actualRow = rows.get(row);
-      List<Types.NestedField> fields = icebergReadSchema.asStruct().fields();
-
-      for (int i = 0; i < fields.size(); i += 1) {
-        Object expectedValue = null;
-        if (i >= writeSchema.getFields().size() && fields.get(i).hasDefaultValue()) {
-          expectedValue = fields.get(i).getDefaultValue();
-        } else if (i < writeSchema.getFields().size()) {
-          expectedValue = expectedRow.get(i);
-        }
-        Type fieldType = fields.get(i).type();
-        Object actualValue = actualRow.isNullAt(i) ? null : actualRow.get(i, convert(fieldType));
-        Assert.assertEquals(expectedValue, actualValue);
-      }
+      final InternalRow expectedFirstRow = new GenericInternalRow(8);
```

Comment:
One last comment, I think we should rename `expectedFirstRow` to just `expectedRow`.